open-banking-accelerator/accelerators/ob-apim/repository/resources/apis/DynamicClientRegistration/dcr-swagger.yaml

```diff
@@ -0,0 +1,370 @@
+      security:
+        - TPPOAuth2Security: []
+    put:
+      summary: Update a Client Registration for a given Client ID
+      tags:
+        - Client Registration
+      parameters:
+        - $ref: '#/parameters/ClientRegistrationRequest'
+        - $ref: '#/parameters/Authorization'
+        - $ref: '#/parameters/ClientId'
+      responses:
+        '200':
+          description: Client registration update success
+          schema:
+            $ref: '#/definitions/RegistrationProperties'
+        '400':
+          $ref: '#/responses/400Error'
+        '401':
+          $ref: '#/responses/401Error'
+        '403':
+          $ref: '#/responses/403Error'
+      x-auth-type: Application
+      x-throttling-tier: Unlimited
+      security:
+        - TPPOAuth2Security: []
+    delete:
+      summary: Delete a Client Registration for a given Client ID
+      tags:
+        - Client Registration
+      parameters:
+        - $ref: '#/parameters/Authorization'
+        - $ref: '#/parameters/ClientId'
+      responses:
+        '204':
+          description: Client deleted
+        '401':
+          $ref: '#/responses/401Error'
+        '403':
+          $ref: '#/responses/403Error'
+        '405':
+          $ref: '#/responses/403Error'
+      x-auth-type: Application
+      x-throttling-tier: Unlimited
+      security:
+        - TPPOAuth2Security: []
+parameters:
+  Authorization:
+    in: header
+    name: Authorization
+    type: string
+    required: true
+    description: 'An Authorisation Token as per [RFC6750](https://tools.ietf.org/html/rfc6750)'
+  ClientId:
+    in: path
+    name: ClientId
+    description: The client ID issued by the target Data Holder
+    type: string
+    required: true
+  ClientRegistrationRequest:
+    in: body
+    name: ClientRegistrationRequest
+    description: 'The registration request JWT, as defined in [Dynamic Client Registration](https://cdr-register.github.io/register/#dynamic-client-registration), to be used to register with a Data Holder'
+    required: true
+    schema:
+      type: string
+      format: JWT
+      example: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...'
+responses:
+  400Error:
+    description: Request failed due to client error
+    schema:
+      $ref: '#/definitions/RegistrationError'
+  401Error:
+    description: Request failed due to unknown or invalid Client or invalid access token
+    headers:
+      WWW-Authenticate:
+        description: 'The Response Header Field as per [RFC6750](https://tools.ietf.org/html/rfc6750)'
+        type: string
+        pattern: ^Bearer .*
+  403Error:
+    description: 'The client does not have permission to read, update or delete the Client'
+  405Error:
+    description: Method Not Allowed
+securityDefinitions:
+  TPPOAuth2Security:
+    type: oauth2
+    flow: application
+    scopes:
+        cdr-registration: "Ability to call registration endpoint"
+    x-scopes-bindings:
+      cdr-registration: "Internal/subscriber"
```

Comment:
can we remove 'cdr'

Reply:
as discussed, these scopes are referred in many places including the implementation. Therefore, I changed the swagger description[1] to indicate this was created referring the AU spec.

[1] - https://github.com/wso2/financial-open-banking/commit/d596ec7445df85e49b1dd15715763d339a0a0730